From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jelle Raaijmakers <jelle@gmta.nl>
Date: Mon, 9 May 2022 01:01:47 +0200
Subject: [PATCH] SDL: Fix 2x audio rate issue

By not passing in an `obtained` struct into `SDL_OpenAudio`, we ask SDL
to perform any sample rate and/or format conversion for us. Previously
GLTron would simply ignore the result in `obtained`, causing the audio
to be played back at 2x speed.
---
 src/audio/sound_glue.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/audio/sound_glue.cpp b/src/audio/sound_glue.cpp
index 9753f09..e08b1b3 100644
--- a/src/audio/sound_glue.cpp
+++ b/src/audio/sound_glue.cpp
@@ -170,7 +170,7 @@ extern "C" {
 
 		SDL_AudioSpec obtained;
 
-    if(SDL_OpenAudio( spec, &obtained ) != 0) {
+    if(SDL_OpenAudio( spec, nullptr ) != 0) {
       fprintf(stderr, "[error] %s\n", SDL_GetError());
       sound->SetStatus(Sound::eUninitialized);
     } else {
